src/holodeck/environments.py

```diff
@@ -730,3 +639,124 @@ def _create_copy(self, obj):
+        will fail silently.
+
+        Args:
+            hour (:obj:`int`): The hour in 24-hour format, between 0 and 23 inclusive.
+        """
+        self.send_world_command("SetHour", num_params=[hour % 24])
+
+    def start_day_cycle(self, day_length):
+        """Start the day cycle.
+
+        The cycle will start when :meth:`tick` or :meth:`step` is called next.
+
+        The sky sphere will then update each tick with an updated sun angle as it moves about the]
+        sky. The length of a day will be roughly equivalent to the number of minutes given.
+
+        If there is no skysphere, skylight, or directional source light in the world, this command
+        will fail silently.
+
+        Args:
+            day_length (:obj:`int`): The number of minutes each day will be.
+        """
+        if day_length <= 0:
+            raise HolodeckException("The given day length should be between above 0!")
+
+        self.send_world_command("SetDayCycle", num_params=[1, day_length])
+
+    def stop_day_cycle(self):
+        """Stop the day cycle.
+
+        The cycle will stop when :meth:`tick` or :meth:`step` is called next.
+
+        By the next tick, day cycle will stop where it is.
+
+        If there is no skysphere, skylight, or directional source light in the world, this command
+        will fail silently.
+        """
+        self.send_world_command("SetDayCycle", num_params=[0, -1])
+
+    def set_weather(self, weather_type):
+        """Set the world's weather.
+
+        The new weather will be applied when :meth:`tick` or :meth:`step` is called next.
+
+        By the next tick, the lighting, skysphere, fog, and relevant particle systems will be
+        updated and/or spawned
+        to the given weather.
+
+        If there is no skysphere, skylight, or directional source light in the world, this command
+        will fail silently.
+
+        ..note::
+            Because this command can affect the fog density, any changes made by a
+            ``change_fog_density`` command before a set_weather command called will be undone. It is
+            recommended to call ``change_fog_density`` after calling set weather if you wish to
+            apply your specific changes.
+
+        In all downloadable worlds, the weather is clear by default.
+
+        If the given type string is not available, the command will not be sent.
+
+        Args:
+            weather_type (:obj:`str`): The type of weather, which can be ``rain`` or ``cloudy``.
+
+        """
+        if not weather_type.lower() in ["rain", "cloudy"]:
+            raise HolodeckException("Invalid weather type " + weather_type)
+
+        self.send_world_command("SetWeather", string_params=[weather_type])
+
+    def _enqueue_command(self, command_to_send):
+        self._command_center.enqueue_command(command_to_send)
+
+    # TODO(vinhowe): This is duplicated from HolodeckEnvironment--figure out where this logic needs to be
```

Comment:
You could just pass `HolodeckEnvironment`'s `send_world_command()` in the init and save it as a class member
```python
 class a:
        def __init__(self):
                self.burrito = b(self.taco)

        def taco(self):
                print("Class a!")

        def test(self):
                self.burrito.do_the_thing()

class b:
        def __init__(self, frobulate):
                self.frobulate = frobulate

        def do_the_thing(self):
                self.frobulate()

c = a()
c.test()
```